src/utils/Subscription.js

```diff
@@ -1,45 +1,61 @@
 // encapsulates the subscription logic for connecting a component to the redux store, as
 // well as nesting subscriptions of descendant components, so that we can ensure the
 // ancestor components re-render before descendants
+
+function initListeners() {
+  let count = 0
+  let current = []
+  let next = []
+
+  return {
+    clear() {
+      count = 0
+      next = null
```

Comment:
Maybe we should clear `next` and `current` to be empty arrays. That way we can always check either's `length` property instead of keeping `count`.